src/main/java/org/websoso/WSSServer/repository/NovelCustomRepositoryImpl.java

```diff
@@ -0,0 +1,74 @@
+package org.websoso.WSSServer.repository;
+
+import static org.websoso.WSSServer.domain.QNovel.novel;
+import static org.websoso.WSSServer.domain.QUserNovel.userNovel;
+
+import com.querydsl.core.types.dsl.BooleanExpression;
+import com.querydsl.core.types.dsl.Expressions;
+import com.querydsl.core.types.dsl.NumberTemplate;
+import com.querydsl.core.types.dsl.StringTemplate;
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import java.util.Collections;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Repository;
+import org.websoso.WSSServer.domain.Novel;
+
+@Repository
+@RequiredArgsConstructor
+public class NovelCustomRepositoryImpl implements NovelCustomRepository {
+
+    private final JPAQueryFactory jpaQueryFactory;
+
+    @Override
+    public Page<Novel> findSearchedNovels(Pageable pageable, String query) {
+
+        if (query.isEmpty() || query.isBlank()) {
+            return new PageImpl<>(Collections.emptyList(), pageable, 0);
+        }
+
+        String searchQuery = query.replaceAll("\\s+", "");
+
+        StringTemplate cleanTitle = Expressions.stringTemplate(
+                "REPLACE(REPLACE({0}, ' ', ''), CHAR(9), '')",
+                novel.title
+        );
+        StringTemplate cleanAuthor = Expressions.stringTemplate(
+                "REPLACE(REPLACE({0}, ' ', ''), CHAR(9), '')",
+                novel.author
+        );
+
+        NumberTemplate<Long> popularity = Expressions.numberTemplate(Long.class,
+                "(SELECT COUNT(un) FROM UserNovel un WHERE un.novel = {0} AND (un.isInterest = true OR un.status <> 'QUIT'))",
+                novel);
+
+        BooleanExpression titleContainsQuery = cleanTitle.containsIgnoreCase(searchQuery);
+        BooleanExpression authorContainsQuery = cleanAuthor.containsIgnoreCase(searchQuery);
+
+        List<Novel> novelsByTitle = jpaQueryFactory.selectFrom(novel)
+                .leftJoin(novel.userNovels, userNovel)
```

Comment:
p3;
```suggestion
        List<Novel> novelsByTitle = jpaQueryFactory
                .selectFrom(novel)
                .leftJoin(novel.userNovels, userNovel)
```

개행하면 더 읽기 편할 것 같아요

Reply:
적용했습니다!!